### DISCUSSION TITLE
✨ support mounting device into the build

### DISCUSSION THREAD

**Discussion by dolanor** [dagger/dagger#12072] [Ideas]
### What are you trying to do?

I need to access `/dev/ttyUSB0` to be able to flash some stm32 microcontroller.
Once I built an app with tinygo, I need to use `stm32flash` on `/dev/ttyUSB0` with it.

### Why is this important to you?

We could simplify the whole flashing process by removing the dependencies on the host to be able to hack on those devices.

### How are you currently working around this?

I do it directly on the host (and have a bunch of microcontroller tools on it)

**Comment by sipsma** [comment]
Not sure if Buildkit exposes the knobs we'd need to include the device in the container today. Could be buried in the oci worker config (in buildkit's toml) somewhere.

The other hurdle will be that buildkitd by default runs containerized (via our docker provisioner), so we'll also need to poke the device nodes through that too.

**Comment by daniel-naegele** [comment]
Hey is there any update on this? This would be really awesome in order to be able to start virtual machines using Qemu for services. Then the `/dev/kvm` device could be passed through the container..
Usually one would of course use docker containers to manage services, but sometimes its just necessary to have a VM. For example running an Android Emulator for automated testing.
